### PR TITLE
Add optimisations to make running Lua tests more tolerable.

### PR DESCRIPTION
**Pull request by ptersilie**
See commits for more info. Bottom line is, this significantly speeds up lua tests, to continue working on correctness.

### CODE REVIEW

File: ykcapi/src/lib.rs [hunk 1]
```diff
@@ -238,6 +238,8 @@ pub extern "C" fn yk_stopgap(
     let mut framerec = unsafe { FrameReconstructor::new(activeframes) };
 
     // Parse the stackmap of the JIT module.
+    // YKFIXME: Parsing the stackmap and initialising `framerec` has a 40ms overhead, which could
```
Comment (ltratt) [inline ykcapi/src/lib.rs:241]: Should this be `YKOPT`?

Comment (vext01) [reply ykcapi/src/lib.rs:241]: Also the 40s metric is specific to yklua. I think it's enough just to say "this is really slow" or something like this.

File: ykfr/src/lib.rs [hunk 2]
```diff
@@ -6,12 +8,32 @@ use std::{
...
 };
-use yksmp::{Location as SMLocation, StackMapParser};
+use yksmp::{Location as SMLocation, SMEntry, StackMapParser};
 
 mod llvmbridge;
 use llvmbridge::{get_aot_original, Module, Type, Value};
 
+pub static AOT_STACKMAPS: LazyLock<Vec<SMEntry>> = LazyLock::new(|| {
+    // Load the stackmap from the binary to parse in the stackmaps.
+    // FIXME: Don't use current_exe.
```
[the remark this answers is not in the document]
Comment (vext01) [reply ykfr/src/lib.rs:20]: The `YK` prefix is only used in ykllvm to distinguish our FIXME from theirs.

File: ykfr/src/lib.rs [hunk 3]
```diff
@@ -6,12 +8,32 @@ use std::{
...
+    let pathb = env::current_exe().unwrap();
+    let file = fs::File::open(&pathb.as_path()).unwrap();
```
Comment (vext01) [inline ykfr/src/lib.rs:22]: There are several places in the system where the stackmap section is being loaded from disk. At some point we should have ykllvm mark the section loadable and fish that data out of the virtual address space.

Can you add a FIXME for this please?

File: ykcapi/src/lib.rs (hunk 1 above)
Comment (ptersilie) [reply ykcapi/src/lib.rs]: Both fixed in [c8af45b](https://github.com/ykjit/yk/pull/619/commits/c8af45b27186f79fa414ac5cffb345b1fd48077c)

File: ykfr/src/lib.rs (hunk 3 above)
Comment (ptersilie) [reply ykfr/src/lib.rs:22]: I'm not aware of other places where we parse the AOT stackmap. The only other place where `StackmapParser` is used is to get the stackmaps of a trace.

**Comment by ptersilie** [comment]
Addressed all comments.

File: ykfr/src/lib.rs (hunk 3 above)
Comment (vext01) [reply ykfr/src/lib.rs:22]: Admittedly I only did a quick grep for the string `".llvm_stackmaps"`, which appears in many places.

**Comment by vext01** [comment]
please squash

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**